pkgs/servers/misc/navidrome/default.nix

```diff
@@ -1,48 +1,40 @@
-{ lib, stdenv, fetchurl, ffmpeg, ffmpegSupport ? true, makeWrapper, nixosTests }:
+{ buildGoModule, fetchFromGitHub, lib, pkg-config, stdenv, ffmpeg, taglib, zlib, ffmpegSupport ? true, makeWrapper, nixosTests }:
 
 with lib;
 
-stdenv.mkDerivation rec {
-  pname = "navidrome";
-  version = "0.47.5";
+buildGoModule {
 
+  pname = "navidrome";
 
-  src = fetchurl (if stdenv.hostPlatform.system == "x86_64-linux"
-  then {
-    url = "https://github.com/deluan/navidrome/releases/download/v${version}/navidrome_${version}_Linux_x86_64.tar.gz";
-    sha256 = "sha256-AkSjtln53HDdIcQgnA8Wj010RXnOlOsFm2wfVgbvwtc=";
-  }
-  else {
-    url = "https://github.com/deluan/navidrome/releases/download/v${version}/navidrome_${version}_Linux_arm64.tar.gz";
-    sha256 = "sha256-+VBRiV2zKa6PwamWj/jmE4iuoohAD6oeGnlFi4/01HM=";
-  });
+  version = "0.47.5";
 
-  nativeBuildInputs = [ makeWrapper ];
+  src = fetchFromGitHub {
+    owner = "navidrome";
+    repo = "navidrome";
+    rev = "v${version}";
+    sha256 = "sha256-gTvJI+brdEpdpbEcdQycqw15seI+k5dMDVrjY3v6i14=";
+  };
 
-  unpackPhase = ''
-    tar xvf $src navidrome
-  '';
+  vendorSha256 = "sha256-xMAxGbq2VSXkF9R9hxB9EEk2CnqsRxg2Nmt7zyXohJI=";
 
-  installPhase = ''
-    runHook preInstall
+  nativeBuildInputs = [ makeWrapper pkg-config ];
 
-     mkdir -p $out/bin
-     cp navidrome $out/bin
+  buildInputs = [ taglib zlib ];
 
-    runHook postInstall
-  '';
+  CGO_CFLAGS = [ "-Wno-return-local-addr" ];
 
   postFixup = optionalString ffmpegSupport ''
     wrapProgram $out/bin/navidrome \
       --prefix PATH : ${makeBinPath [ ffmpeg ]}
   '';
 
+  doCheck = false;
+
   passthru.tests.navidrome = nixosTests.navidrome;
 
   meta = {
     description = "Navidrome Music Server and Streamer compatible with Subsonic/Airsonic";
     homepage = "https://www.navidrome.org/";
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
     license = licenses.gpl3Only;
     platforms = [ "x86_64-linux" "aarch64-linux" ];
```

Comment:
Maybe now we can support more platforms?

Reply:
Expanded to platforms.linux

Reply:
Why not `lib.platforms.unix`?

Reply:
Changed to `platforms.unix`